src/rpc/server.cpp

```diff
@@ -490,7 +491,42 @@ UniValue CRPCTable::execute(const std::string &strMethod, const UniValue &params
     try
     {
         // Execute
-        return pcmd->actor(params, false);
+        auto paramRange = rpcCvtTable.find(strMethod);
+        if (paramRange != rpcCvtTable.end()) {
+            auto numRequired = paramRange->second.first.size();
+            auto numOptional = paramRange->second.second.size();
+            if (params.size() < numRequired || numRequired + numOptional < params.size()) {
+                std::string helpMsg;
+                try {
+                    // help gets thrown – if it doesn’t throw, then no help message
+                    pcmd->actor(params, true);
+                } catch (const std::runtime_error& err) {
+                    helpMsg = std::string("\n\n") + err.what();
+                }
+                throw JSONRPCError(
+                    RPC_INVALID_PARAMS,
```

Comment:
Turns out that this change broke https://github.com/s-nomp/node-stratum-pool because it was depending on `RPC_MISC_ERROR` code being returned when an individual method returned help text due to an invalid number of parameters.

I encountered this because I noticed that zcashd and Zallet returned different error codes in this case, and then found that our custom code mapping from `ErrorCode::InvalidParams` to `LegacyCode::Misc` was originally introduced into Zebra to get around `jsonrpsee` returning the former for an invalid number of parameters (https://github.com/ZcashFoundation/zebra/issues/6054).

So I think we were incorrect to change this error code, and this should have been:
```suggestion
                    RPC_MISC_ERROR,
```

I also believe that the Zebra fix was incorrect. They universally apply the map from `ErrorCode::InvalidParams` to `LegacyCode::Misc`, whereas in zcashd `RPC_MISC_ERROR` is only used when a `std::exception` is thrown from an RPC method handler, not any other `JSONRPCError` (and most invalid parameters are reported via that with `RPC_INVALID_PARAMETER` which is a legacy Bitcoin error code).